tools/dds/dds-sniffer/dds-sniffer.cpp

```diff
@@ -300,3 +326,251 @@ void Sniffer::on_type_information_received( eprosima::fastdds::dds::DomainPartic
 {
     std::cout << "Topic '" << topic_name << "' with type '" << type_name << "' received" << std::endl;
 }
+
+void Sniffer::save_topic_writer( const eprosima::fastrtps::rtps::WriterDiscoveryInfo & info )
+{
+    std::string topic_name = info.info.topicName().c_str();
+    if( _discoveredTopics.find( topic_name ) == _discoveredTopics.end() )
+    {
+        _discoveredTopics.insert( { topic_name, *std::make_shared< ReadersWriters >() } );
```

Comment:
`*std::make_shared< ReadersWriters >()`?
Doesn't line 338 already do all this?

Reply:
Line 335 creates and inserts a new topic map entry, line 338 inserts writer information to the writers set of that map entry

Reply:
It is indeed redundant and will be removed. Map operator[] creates entry if does not exist